compiler/resolution/functionResolution.cpp

```diff
@@ -5151,9 +5151,12 @@ static void resolveInitVar(CallExpr* call) {
       targetType = useType;
     }
 
-    // Ignore the target type if it's the same & not a runtime type.
+    // Ignore the target type if it's the same & not a runtime type
+    // and not sync/single (since sync/single initCopy returns a different type)
     if (targetType == srcType &&
-        !targetType->symbol->hasFlag(FLAG_HAS_RUNTIME_TYPE)) {
+        !targetType->symbol->hasFlag(FLAG_HAS_RUNTIME_TYPE) &&
+        !isSyncType(srcType) &&
+        !isSingleType(srcType)) {
```

Comment:
Man, I coulda sworn we'd already done that.  Did it get accidentally removed in the other PR?

Reply:
We did it in the code handling PRIM_COERCE and also lower down here but not in this specific spot.

Reply:
Ah, okay